src/dump.c

```diff
@@ -499,12 +499,13 @@ static int type_recursively_external(jl_datatype_t *dt)
 
     int i, l = jl_svec_len(dt->parameters);
     for (i = 0; i < l; i++) {
-        jl_datatype_t *p = (jl_datatype_t*)jl_tparam(dt, i);
+        jl_value_t *p0 = jl_tparam(dt, i);
+        jl_datatype_t *p = (jl_datatype_t*)jl_unwrap_unionall(p0);
```

Comment:
can't use `unwrap_unionall` here: need to recursively check that all of the bounds of the typevars are also `type_recursively_external` (this is a part of an optimization that needs to detect whether the type only contains references to external objects and nothing created here)